server/test/modules/provisioning/infrastructureConfigurationProvider.spec.js

```diff
@@ -0,0 +1,174 @@
+    mapStubs['../data-access/services'] = deps.servicesDbMock;
+    var infrastructureConfigurationProvider = proxyquire('../../../modules/provisioning/infrastructureConfigurationProvider', mapStubs);
+
+    var target = infrastructureConfigurationProvider;
+
+    return target.get(input.environmentName, input.serviceName, input.serverRoleName);
+  }
+
+  describe("Getting the configuration for a service on an environment", () => {
+
+    it("Should return configuration error when no service is found", () => {
+      const expectedError = `An error has occurred retrieving "AService" service: Service "AService" not found.`;
+      setupDependencies();
+      const nullService = null;
+      deps.servicesDbMock.get = sinon.stub().returns(Promise.resolve(nullService));
+
+      var promise = getInstance();
+
+      return promise.catch(configurationError => {
+        configurationError.should.have.property('message', expectedError);
+      });
+    });
+
+
+    it("Should throw error when server role doesn't have owning cluster", () => {
+      var expectedError = `An error has occurred retrieving the serverRole, no owning cluster found "${JSON.stringify(serverRole)}".`;
+      setupDependencies();
+      delete serverRole.OwningCluster;
+      var promise = getInstance();
+
+      return promise.catch(error => {
+        error.should.have.property('message', expectedError);
+      });
+    });
+
+    it("Should throw error when no environment type is found", () => {
+      const exception = "Bang!!!";
+      setupDependencies();
+      deps.environmentTypeMock.getByName = sinon.stub().returns(Promise.reject(exception));
+      var expectedError = `An error has occurred retrieving environment "${input.environmentName}" environment type.`;
+
+      var promise = getInstance();
+
+      return promise.catch(error => {
+        error.should.have.property('message', expectedError);
+        error.should.have.property('innerError', exception);
+      });
+    });
+
+    it("Should throw error when cluster can't be found on dynamo", () => {
+      setupDependencies();
+      deps.clustersMock.get = sinon.stub().returns(Promise.reject(new DynamoItemNotFoundError("", "")));
+      var expectedError = `Cluster "Ransom" not found.`;
+
+      var promise = getInstance();
+
+      return promise.catch(error => {
+        error.should.have.property('message', expectedError);
+      });
+    });
+
+    it("Should throw error when can't get cluster from dynamo", () => {
+      setupDependencies();
+      deps.clustersMock.get = sinon.stub().returns(Promise.reject({ message: "Bang!" }));
+      var expectedError = `An error has occurred retrieving "Ransom" cluster: ${"Bang!"}`;
+
+      var promise = getInstance();
+
+      return promise.catch(error => {
+        error.should.have.property('message', expectedError);
+      });
+    });
+
+    it("Should return a configuration object", () => {
+      setupDependencies();
```

Comment:
setupDependencies can be moved to a beforeEach

Reply:
👍